### ISSUE TITLE
FARTTY_INPUT_MOUSE wheel events

### ISSUE THREAD

**Issue by magiblot** [elfmz/far2l#1501]
Hello,

According to FarTTY.h, one of the arguments included in FARTTY_INPUT_MOUSE notifications is the `dwButtonState` field of a `MOUSE_EVENT_RECORD` struct.

In the [Win32 API](https://learn.microsoft.com/en-us/windows/console/mouse-event-record-str), this field is specified as follows:

* In `MOUSE_WHEELED` events, if the high word of the `dwButtonState` member contains a positive value, the wheel was rotated forward, away from the user.
* In `MOUSE_HWHEELED` events, If the high word of the `dwButtonState` member contains a positive value, the wheel was rotated to the right. Otherwise, the wheel was rotated to the left.

In other words, one can check the scroll direction by testing whether bit 31 of `dwButtonState` is set.

However, in events sent by far2l, bit 16 of `dwButtonState` is set instead. For example, when the mouse wheel is scrolled forward, the value of `dwButtonState` is `0x00010000` even though I was expecting a `0x80000000`.

Therefore, I ask you: is this behaviour intended, or is it a bug?

Thank you very much.

**Comment by elfmz** [comment]
Looks like a bug, actually such behavior only in WX backend, in TTY high word is set to 0xffff/0x0001 but not 0x0000/0x0001
Internally far2l checks value for being > 0 here - https://github.com/elfmz/far2l/blob/master/far2l/src/console/keyboard.cpp#L1246
So i suggest for compatibility to do same check for greater zero if need to parse incoming data but do 0xffff/0x0001 if need to produce data. I will also commit change into far2l's wx backend implementation so it will also produce 0xffff/0x0001 just to fit the documentation.

**Comment by unxed** [comment]
Source bug https://github.com/magiblot/turbo/issues/53 not reproduced any more.

**Comment by elfmz** [comment]
closing this too then

**Comment by magiblot** [comment]
Great. Thank you both.
